packages/orm/src/QueryBuilder.ts

```diff
@@ -318,9 +341,16 @@ export function buildQuery<T extends Entity>(
         const serde = field.serde!;
         // TODO Currently hardcoded to single-column support; poly is handled above this
         const column = serde.columns[0];
-        // TODO Currently we only support base-type WHEREs if the sub-type is the main `em.find`
-        // const maybeBaseAlias = field.alias;
-        query = hasClause ? addPrimitiveClause(query, alias, column, clause) : query;
+        // If the field is a base type field we add the clause to the base type table
+        const maybeBaseType = baseTypeFields[key];
+        const baseMeta = meta.baseTypes.find((b) => b.type === maybeBaseType)!;
+        if (maybeBaseType && hasClause && baseMeta) {
+          const baseTypeIndex = meta.baseTypes.indexOf(baseMeta);
+          const baseAlias = `b${baseTypeIndex}`;
+          addClauses(baseMeta, baseAlias, baseTypeTree[maybeBaseType] as any, order);
```

Comment:
Curious to test this with something like this:

```ts
/**
  Where the Animal type will be something like:
  type Animal {
    name: String!
  }
*/


const human = newHuman(em, { name: "Jane" }); // Inherits from Animal
const dog = newDog(em, { name: "Zeus", owner: human }); // Also inherits from Animal

const result = await em.findOne(Dog, { name:"Zeus", owner: { name: "Jane" } });
```

Maybe worth testing this to see if my approach with the object-trees works or if it needs some tweaks.